### PR TITLE
DHFPROD-2560: Prototype of an installer for local and DHS environments

### PR DESCRIPTION
**Pull request by rjrudin**
I'd like to get this merged into develop so it's easier for the DHS team to build, but it can wait until after 5.0.1 - though it has no impacts on any code, it's all new stuff in the "hub.cli" package. 

Instructions for using the installer are on our internal Wiki.

### CODE REVIEW

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/AbstractInstallerCommand.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.hub.impl.HubConfigImpl;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Properties;
+
+public abstract class AbstractInstallerCommand extends LoggingObject implements InstallerCommand {
+
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/cli/AbstractInstallerCommand.java]: protected HubConfigImpl hubConfig;

public AbstractInstallerCommand(HubConfigImpl hubConfig) {
      this. hubConfig = hubConfig;
}

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/AbstractVerifyCommand.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+import com.marklogic.mgmt.resource.security.AmpManager;
+import com.marklogic.mgmt.resource.security.PrivilegeManager;
+import com.marklogic.mgmt.resource.triggers.TriggerManager;
+import com.marklogic.rest.util.Fragment;
+import com.marklogic.rest.util.ResourcesFragment;
+import org.springframework.util.Assert;
+
+public abstract class AbstractVerifyCommand extends AbstractInstallerCommand {
+
+    protected HubConfigImpl hubConfig;
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/cli/AbstractVerifyCommand.java]: removed protected HubConfigImpl hubConfig;

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/AbstractVerifyCommand.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+    public AbstractVerifyCommand(HubConfigImpl hubConfig) {
+        this.hubConfig = hubConfig;
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/cli/AbstractVerifyCommand.java]: changed to super(hubConfig) ?

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/InstallLocalDhfCommand.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+@Parameters(commandDescription = "Install or upgrade DHF into a local (non-DHS) environment")
+public class InstallLocalDhfCommand extends AbstractInstallerCommand {
+
+    private ObjectMapper objectMapper = ObjectMapperFactory.getObjectMapper();
+
+    private DataHubImpl dataHub;
+    private HubConfigImpl hubConfig;
+
+    public InstallLocalDhfCommand(DataHubImpl dataHub, HubConfigImpl hubConfig) {
+        this.dataHub = dataHub;
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/cli/InstallLocalDhfCommand.java]: super(hubConfig);
this.dataHub = dataHub;

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/InstallLocalDhfCommand.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import java.io.IOException;
+import java.util.Map;
+
+@Parameters(commandDescription = "Install or upgrade DHF into a local (non-DHS) environment")
+public class InstallLocalDhfCommand extends AbstractInstallerCommand {
+
+    private ObjectMapper objectMapper = ObjectMapperFactory.getObjectMapper();
+
+    private DataHubImpl dataHub;
+    private HubConfigImpl hubConfig;
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/cli/InstallLocalDhfCommand.java]: https://github.com/marklogic/marklogic-data-hub/blob/dae30aca47ce12b7bc934740ee1664d1acb73ae1/marklogic-data-hub/src/main/java/com/marklogic/hub/cli/InstallLocalDhfCommand.java#L25

could remove the line

File: marklogic-data-hub/src/main/java/com/marklogic/hub/cli/InstallDhfInDhsCommand.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    )
+    private String hostname;
+
+    private DataHubImpl dataHub;
+    private HubConfigImpl hubConfig;
+
+    public InstallDhfInDhsCommand(DataHubImpl dataHub, HubConfigImpl hubConfig) {
+        this.dataHub = dataHub;
+        this.hubConfig = hubConfig;
+    }
```
Comment (hao1st) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/cli/InstallDhfInDhsCommand.java]: same suggestion as AbstractVerifyCommand

**Comment by rjrudin** [comment]
Closing for now as it's WIP

**Pull request closed without merging**